dot/parachain/backing/validated_candidate_command.go

```diff
@@ -268,12 +268,12 @@ func (cb *CandidateBacking) handleCommandAttest(
 		return nil
 	}
 
-	statement := parachaintypes.NewStatementVDT()
-	if err := statement.SetValue(parachaintypes.Valid(candidateHash)); err != nil {
-		return fmt.Errorf("setting statement: %w", err)
-	}
-
 	if bgValidationResult.err == nil {
+		statement := parachaintypes.NewStatementVDT()
+		if err := statement.SetValue(parachaintypes.Valid(candidateHash)); err != nil {
```

Comment:
I would say we should panic if we cannot set a VDT value. The point is:
in rust if you have the following enum:

```rs
enum Example {
  A, B, C
}

// and try to assign
let a: Example = Example::D
```
this will lead to a compile time error, since D is not defined in the Example enum, and we should have the same behavior. 

afaik we don't have any recovery plan if such error happens, so panicking the node is a better approach since this might be a development problem and not a network problem.

Reply:
I agree. 
But we have not used panic for VDT in Parachain. 

Can we make this change in a separate PR so panic can be used everywhere in parachain?